crates/beerus-core/src/lightclient/beerus.rs

```diff
@@ -298,17 +298,36 @@ impl BeerusLightClient {
         &self,
         contract_address: FieldElement,
         storage_key: FieldElement,
+        block_id: &BlockId,
     ) -> Result<FieldElement> {
-        let last_block = self
+        let last_proven_block = self
             .ethereum_lightclient
             .read()
             .await
             .starknet_last_proven_block()
             .await?
             .as_u64();
-        self.starknet_lightclient
-            .get_storage_at(contract_address, storage_key, last_block)
-            .await
+
+        if let BlockId::Number(block_number) = block_id {
+            if block_number <= &last_proven_block {
+                return self
+                    .starknet_lightclient
+                    .get_storage_at(contract_address, storage_key, block_id)
+                    .await;
+            }
+        } else if let MaybePendingBlockWithTxHashes::Block(block) = self
+            .starknet_lightclient
+            .get_block_with_tx_hashes(block_id)
+            .await?
+        {
+            if block.block_number <= last_proven_block {
+                return self
+                    .starknet_lightclient
+                    .get_storage_at(contract_address, storage_key, block_id)
+                    .await;
+            }
+        }
+        Err(eyre::eyre!("BlockId is not proven yet"))
```

Comment:
Here I am ensuring that the `blockId` received as a parameter is validated by L1 and we are also following the StarkNet Specs. What do you think @phklive

Reply:
From what we have talked @dpinones, looks like we are doing too much validations that are also done by the full node under our feet.

I think @phklive had other idea in mind with this comment:

    We need to keep using this one, using the last block from ethereum makes sure that the data that we return is valid and safe. 
    If not a starknet node could lie considering that the data would not be validated on L1 yet.
    
Or @phklive what do you think? We should keep the way @dpinones implemented it?